### PR TITLE
Docs 56 restore missing doc files

### PR DESCRIPTION
**Pull request by igorwojciechowski**
### Related Issues

closes DOCS-56

### Short Description of the PR

Restores missing docs due to release branch merge.

### Pull Request Checklist

- [ ] I have updated the Changelog ([V1](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md)) [v2](https://docs-next.vuestorefront.io/contributing/creating-changelog.html) and mentioned all breaking changes in the public API.
- [ ] I have documented all new public APIs and made changes to existing docs mentioning the parts I've changed so they're up to date.
- [ ] I have tested my Pull Request on production build and (to my knowledge) it works without any issues

- I tested manually my code and it works well with both:
- [ ] Default Theme
- [ ] Capybara Theme
- [ ] I have written test cases for my code

- [ ] I have followed [naming conventions](https://github.com/kettanaito/naming-cheatsheet)

### CODE REVIEW

**Review by lukaszjedrasik — COMMENTED** [review]
Looks good. Just change the target branch to `release-2.4`

File: README.md [hunk 1]
```diff
@@ -6,8 +6,7 @@
 
 Vue Storefront is the most popular and most advanced Frontend Platform for eCommerce
 - [Documentation](https://docs.vuestorefront.io/v2/)
-- [Demo](https://vsf-next-demo.storefrontcloud.io/)
-- [Installation](https://docs.vuestorefront.io/v2/general/installation.html)
```
Comment (filipsobol) [inline README.md:10]: Installation link should stay

**Review by filipsobol — APPROVED** [review]

**Review by lukaszjedrasik — APPROVED** [review]

**Review by lukaszjedrasik — CHANGES_REQUESTED** [review]
Zrzut ekranu 2021-08-16 o 16 29 45

@igorwojciechowski I think we have missing `Guides` section

**Review by lukaszjedrasik — APPROVED** [review]

File: packages/core/docs/advanced/server-middleware.md [hunk 2]
```diff
@@ -1,12 +1,16 @@
...
+:::
+
 ## What is Vue Storefront Middleware and why we need it?
 
 The Vue Storefront middleware is an Express proxy that takes the requests from the front-end, translates them to a given integration, and calls related API-client.
 
 We have implemented it for a variety of reasons.
 
-First of all, it allows us to provide a proven way of extensibility. As a developer, you have control of the requests and responses in the given platform with [extensions](/advanced/server-middleware.html#extending-middleware))
+First of all, it allows us to provide a proven way of extensibility. As a developer, you have control of the requests and responses in the given platform with [extensions](/advanced/server-middleware.html#extending-integrations))
```
Comment (filipsobol) [inline packages/core/docs/advanced/server-middleware.md:13]: I might be wrong, but shouldn't the link end with `.html`?

Comment (igorwojciechowski) [reply packages/core/docs/advanced/server-middleware.md:13]: But it does, doesn't it? ;o Before `#`?

Comment (filipsobol) [reply packages/core/docs/advanced/server-middleware.md:13]: Missed it, sorry :)

**Pull request merged**